zarr/core.py

```diff
@@ -1176,3 +1176,7 @@ def view(self, shape=None, chunks=None, dtype=None,
             a._filters = filters
 
         return a
+
+    def astype(self, dtype):
+        filt = AsType(dtype=self._dtype, astype=dtype)
+        return self.view(filters=[filt], dtype=dtype)
```

Comment:
This will work if self has no filters, but I think would need to check if there are any filters on the self array, and if so make a new filters list by copying the existing list and prepending the new AsType filter.